eng/testing/debug-dump-template.md

```diff
@@ -0,0 +1,122 @@
+2. Load the dump with a recent WinDbg.
+3. Then run the following commands:
+
+```script
+!setclrpath <path to core root or testhost where coreclr is>
+!sethostruntime '<path to your local dotnet runtime or testhost>'
+.sympath+ <directory with symbols (for library tests these are in testhost dir)>
+```
+### Analyze with dotnet-dump
+
+1. Install [dotnet-dump global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-dump).
+2. Run: `dotnet-dump analyze <path-to-dump>`
+3. Then run the following commands:
+
+```script
+setclrpath <you should see `Load path for DAC/DBI: '<none>'`>
+setclrpath <path to core root or testhost where coreclr is>
+setsymbolserver -directory <directory with symbols (for library tests these are in testhost dir)>
+```
+
+## Linux dumps on windows
+
+In order to debug a Linux dump on Windows, you will have to first go to the PR/CI build
+that sent the test run and download the cross DAC.
+
+Go to your build by navigating to [your build](https://dnceng.visualstudio.com/public/_build/results?buildId=%BUILDID%&view=artifacts&type=publishedArtifacts) and download the `CoreCLRCrossDacArtifacts`, then choose the cross DAC you need depending on the flavor.
+
+Once you've downloaded that artifact, make sure to extract it in the same location where coreclr binary is.
+
+### Debug with WinDbg
+
+1. Install [dotnet-sos global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos).
+2. Load the dump with a recent WinDbg.
+3. Then run the following commands:
+
+```script
+.load '<path to sos (can get it by running `dotnet-sos install`)>'
+!setclrpath <path to core root or testhost where coreclr is>
+!sethostruntime '<path to your local dotnet runtime>'
+.sympath+ <directory with symbols (for library tests these are in testhost dir)>
+```
+### Analyze with dotnet-dump
+
+1. Install [dotnet-dump global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-dump).
+2. Run: `dotnet-dump analyze <path-to-dump>`
+3. Then run the following commands:
+
+```script
+setclrpath <you should see `Load path for DAC/DBI: '<none>'`>
+setclrpath <path to core root or testhost where coreclr is>
+setsymbolserver -directory <directory with symbols (for library tests these are in testhost dir)>
+```
+
+## Linux dumps on Linux
+
+### Debug with LLDB
+
+1. Install [dotnet-sos global tool](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-sos).
+2. Load the dump by running `lldb -c <path-to-dmp> <host binary used (found in testhost)>`
+3. Run the following commands:
+
+```script
+setclrpath <path to core root or testhost where coreclr is>
+sethostruntime '<path to your local dotnet runtime or testhost where coreclr is>'
+setsymbolserver -directory <directory with symbols (for library tests these are in testhost dir)>
+# On linux (native symbols are loaded separately on lldb using `target symbols add <symbol file>`)
```

Comment:
The `loadsymbols` command should also try to find/download/load the native module's symbols.

Reply:
This is new to me. Thanks Mike. I usually had to load coreclr's separately for some reason.